react-native-purchases-ui/src/index.tsx

```diff
@@ -18,11 +28,36 @@ export function presentPaywallIfNeeded(requiredEntitlementIdentifier: String) {
   RNPaywalls.presentPaywallIfNeeded(requiredEntitlementIdentifier);
 }
 
-const ComponentName = 'Paywall';
-
-export const Paywall =
-  UIManager.getViewManagerConfig(ComponentName) != null
-    ? requireNativeComponent(ComponentName)
+const InternalPaywall =
+  UIManager.getViewManagerConfig('Paywall') != null
+    ? requireNativeComponent('Paywall')
     : () => {
       throw new Error(LINKING_ERROR);
     };
+
+export const Paywall: React.FC<FlexPaywallFooterViewProps> = ({style, children}) => (
+  <InternalPaywall style={[{flex: 1}, style]}/>
+);
+
+const InternalPaywallFooterView = UIManager.getViewManagerConfig('Paywall') != null
+  ? requireNativeComponent('RCPaywallFooterView')
+  : () => {
+    throw new Error(LINKING_ERROR);
+  };
+
+type FlexPaywallFooterViewProps = {
+  style?: StyleProp<ViewStyle>;
+  children?: ReactNode;
+};
+
+export const PaywallFooterContainerView: React.FC<FlexPaywallFooterViewProps> = ({style, children}) => (
+  <View style={[{flex: 1}, style]}>
+    <ScrollView
+      style={{backgroundColor: '#4b72f6'}}
+      contentContainerStyle={{flexGrow: 1, paddingBottom: 20}}
+    >
+      {children}
+    </ScrollView>
+    <InternalPaywallFooterView style={{marginTop: -20}}/>
```

Comment:
Why 20 here and above?

Reply:
To overlap the footer with the scrollview. I used 20 because after testing it, that was the minimum I could find it would fit nicely. 20 is actually the radius of the corners in Android and it's pixel independent.